native/ssz_nif/src/utils/from_elx.rs

```diff
@@ -63,3 +66,48 @@ where
         Self::new(value.into_iter().map(FromElx::from).collect()).unwrap()
     }
 }
+
+impl<'a> FromElx<Binary<'a>> for ExecutionBlockHash {
+    fn from(value: Binary<'a>) -> Self {
+        ExecutionBlockHash::from_ssz_bytes(&value).unwrap()
+    }
+}
+
+impl<'a> FromElx<Binary<'a>> for H160 {
+    fn from(value: Binary<'a>) -> Self {
+        H160::from_ssz_bytes(&value).unwrap()
+    }
+}
+
+impl<'a> FromElx<Binary<'a>> for U256 {
+    fn from(value: Binary<'a>) -> Self {
+        U256::from_ssz_bytes(&value).unwrap()
+    }
+}
+
+impl<'a> FromElx<u64> for U256 {
+    fn from(value: u64) -> Self {
+        U256::try_from(value).unwrap()
+    }
+}
+
+impl<'a> FromElx<Binary<'a>> for VariableList<u8, <MainnetEthSpec as EthSpec>::MaxExtraDataBytes> {
+    fn from(value: Binary<'a>) -> Self {
+        VariableList::from_ssz_bytes(&value).unwrap()
+    }
+}
+
+impl<'a> FromElx<Vec<u8>> for VariableList<u8, <MainnetEthSpec as EthSpec>::MaxExtraDataBytes> {
+    fn from(value: Vec<u8>) -> Self {
+        VariableList::from_ssz_bytes(&value).unwrap()
+    }
+}
```

Comment:
If you change this to something similar to the `FromElx<Vec<Elx>> for FixedVector<Lh, N>` implementation, I think it might also work for the `Vec<Vec<u8>>` case.

```rust
impl<'a, Elx, Lh, N> FromElx<Vec<Elx>> for VariableList<Lh, N>
where
    Lh: FromElx<Elx>,
    N: Unsigned,
```

With that, `VariableList<u8>` implements `FromElx<Vec<u8>>`. Then, since `Lh = VariableList<u8>` and `Elx = Vec<u8>` satisfy `Lh: FromElx<Elx>`, `VariableList<Lh>` implements `From<Vec<Elx>>` automatically.

Reply:
I don't think that it would work because the implementation is for Vec<Vec<u8>>